modules/snooker/src/main/ts/ephox/snooker/resize/Sizes.ts

```diff
@@ -41,96 +41,59 @@ const convert = (cell: SugarElement, number: number, getter: (e: SugarElement) =
   return newSize;
 };
 
-const normalizePixelSize = (value: string, cell: SugarElement, getter: (e: SugarElement) => number, setter: (e: SugarElement, value: number) => void): number => {
-  const number = parseInt(value, 10);
+const normalizePixelSize = (value: string, cell: SugarElement<HTMLTableCellElement>, getter: SizeGetter, setter: SizeSetter): number => {
+  const number = parseFloat(value);
   return Strings.endsWith(value, '%') && SugarNode.name(cell) !== 'table' ? convert(cell, number, getter, setter) : number;
 };
 
-const getTotalHeight = (cell: SugarElement): number => {
+const getTotalHeight = (cell: SugarElement<HTMLTableCellElement>): number => {
   const value = getHeightValue(cell);
   if (!value) {
     return Height.get(cell);
   }
   return normalizePixelSize(value, cell, Height.get, setHeight);
 };
 
-const get = (cell: SugarElement, type: 'rowspan' | 'colspan', f: (e: SugarElement) => number): number => {
+const get = (cell: SugarElement<HTMLTableCellElement>, type: 'rowspan' | 'colspan', f: (e: SugarElement) => number): number => {
   const v = f(cell);
   const span = getSpan(cell, type);
   return v / span;
 };
 
-export const getRawWidth = (element: SugarElement): Optional<string> => {
-  // Try to use the style width first, otherwise attempt to get attribute width
-  const cssWidth = Css.getRaw(element, 'width');
+const getRaw = (element: SugarElement<HTMLElement>, prop: 'height' | 'width'): Optional<string> => {
+  // Try to use the style first, otherwise attempt to get the value from an attribute
+  const cssWidth = Css.getRaw(element, prop);
   return cssWidth.fold(() => {
```

Comment:
Feel free to ignore as this is mostly out of scope, but this could be `.orThunk` instead

Reply:
I had to make changes anyways, so went ahead and fixed this as well.